src/irmin/tree.ml

```diff
@@ -1190,14 +1190,14 @@ module Make (P : S.PRIVATE) = struct
   let export ?clear repo contents_t node_t n =
     let seen = seen_of_repo repo in
     let add_node n v () =
-      P.Node.add node_t v >|= fun k ->
+      P.Node.add node_t v >|= fun (k, _) ->
```

Comment:
You should use that new information to stop the recursion in the children node. e.g. if `Created true` then to not call `add_to_todo` on `n` children.

Reply:
hum actually not sure this patch helps very much, as `Node.add` of the parent has to be done after the `Node.add` of the children...